### DISCUSSION TITLE
Document indexing notebook is 404

### DISCUSSION THREAD

**Discussion by IainD925** [microsoft/AI-Chat-App-Hack#141] [General]
[Data ingestion page](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/docs/data_ingestion.md) has link to [this notebook](https://github.com/Azure/azure-search-vector-samples/blob/main/demo-python/code/azure-search-integrated-vectorization-sample.ipynb) with additional detail for document indexing but the notebook page is 404.

The _other samples_ page says JSON document indexing **is not** part of existing functionality:
![NoJSON](https://github.com/microsoft/AI-Chat-App-Hack/assets/98359530/7dff38bd-b670-4ddc-94a5-633d2e51fb42)

Although if you present JSON document **_azure-search-openai-demo_** is capable of indexing them however it doesn't create an index using the JSON structure (even if only JSON documents of the same structure are present) instead it creates the default index with 2 fields (content & embeddings).

@pamelafox is my understanding incorrect (based on reading of [how to index JSON blobs](https://learn.microsoft.com/en-us/azure/search/search-howto-index-json-blobs) ) .... in that the index should create fields as per JSON structure?  Can you advise which file(s) index creation is found?

**Comment by pamelafox** [comment]
Sorry, I'll fix the link, that notebook is now at: https://github.com/Azure/azure-search-vector-samples/blob/main/demo-python/code/integrated-vectorization/azure-search-integrated-vectorization-sample.ipynb

Our RAG app indexes JSON by chunking them and storing chunks in the content field. That works fairly well from our tests (and for the customer who that was originally written for). However, there may be situations where you'd also want to store metadata or structure, or you want to specifically map one JSON to an index. If you went that route, you'd need to decide how you were going to translate user queries into searches on more complex structure, however. I present one approach here:
https://blog.pamelafox.org/2024/03/rag-techniques-using-function-calling.html
